guides/plugins/apps/tax-provider.md

```diff
@@ -0,0 +1,140 @@
+
+In the following, we will have a look at a working example of a tax provider endpoint.
+
+Our implementation uses the [Shopware AppTemplate](https://github.com/shopware/AppTemplate): An easy app-server-integration for Symfony PHP implementations.
+
+{% code title="ProcessController.php" %}
+
+```php
+<?php declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\Shop\ShopRepository;
+use Shopware\AppBundle\Authentication\RequestVerifier;
+use Shopware\AppBundle\Authentication\ResponseSigner;
+use Symfony\Bridge\PsrHttpMessage\Factory\HttpFoundationFactory;
+use Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class ProcessController extends AbstractController
+{
+    public function __construct(
+        private HttpMessageFactoryInterface $psrHttpFactory,
+        private RequestVerifier $requestVerifier,
+        private ResponseSigner $responseSigner,
+        private ShopRepository $shopRepository,
+    ) {
+    }
+
+    #[Route('/process', name: 'process')]
+    public function process(Request $request): Response
+    {
+        $content = \json_decode($request->getContent(), true);
+
+        // shop id is in source->shopId
+        $shopId = $content['source']['shopId'];
+        $shop = $this->shopRepository->getShopFromId($shopId);
+
+        // transform symfony request to psr request
+        // mandatory, if you use the shopware app template
+        // and want to authenticate the requests easily
+        $psrRequest = $this->psrHttpFactory->createRequest($request);
+
+        // authenticate, that the request came from shopware
+        $this->requestVerifier->authenticatePostRequest($psrRequest, $shop);
+
+        $lineItems = $content['cart']['lineItems'];
+        $lineItemTaxes = [];
+
+        // generally, you may want to call a tax provider here instead
+        // for our example we simply assume a hefty tax rate of 50%
+        foreach ($lineItems as $lineItem) {
+            $taxRate = 50;
+            $tax = $lineItem['price']['totalPrice'] * $taxRate / 100;
+
+            $lineItemTaxes[$lineItem['id']] = [
+                [
+                    'tax' => $tax,
+                    'taxRate' => $taxRate,
+                    'price' => $lineItem['price']['totalPrice'],
+                ],
+            ];
+        }
+
+
+        // you can provide lineItemTaxes, deliveryTaxes and cartPriceTaxes
+        // if you do not provide cartPriceTaxes, Shopware will recalculate them according to your provided taxes
+        $responseContent = [
+            'lineItemTaxes' => $lineItemTaxes,
+            //'deliveryTaxes' => [],
```

Comment:
What will be the data layout be for delivery taxes? Are they keyed by deliveryId?